### PR TITLE
[#24] [Integrate] As a user, I can see the survey questions

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/24

## What happened 👀

- Integrate Survey Question screen
- Write unit test for `SurveyQuestionViewModel`

## Insight 📝

We are facing an error when running UITest, the app keeps crashing at `DIkt.doInit` which works normally when running the app
CleanShot 2023-05-08 at 14 23 46
I will investigate this error later and update the UITest

## Proof Of Work 📹

https://user-images.githubusercontent.com/17875522/236765546-c4f6754a-5922-4c82-8de5-5d35dc5cb3f5.mp4

### CODE REVIEW

File: ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/Models/SurveyDetailUIModel.swift [hunk 1]
```diff
@@ -14,18 +14,21 @@ struct SurveyDetailUIModel: Equatable {
     let title: String
     let description: String
     let coverImageURL: URL?
+    let detail: SurveyDetail?
```
Comment (suho) [inline ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/Models/SurveyDetailUIModel.swift]: Don't need to have this data here

File: ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/SurveyDetailScreen.swift [hunk 2]
```diff
@@ -29,7 +30,15 @@ struct SurveyDetailScreen: View {
...
-                        // TODO: Begin survey
+                        guard let surveyDetail = viewModel.survey.detail else { return }
+                        let viewModel = SurveyQuestionViewModel(
+                            surveyDetail: surveyDetail,
+                            questionIndex: 0
+                        )
+                        navigator.show(
+                            screen: .surveyQuestion(viewModel: viewModel),
+                            by: .presentCover
+                        )
```
Comment (suho) [inline ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/SurveyDetailScreen.swift]: let get the `SurveyQuestionViewModel` from viewModel.
```suggestion
                        let surveyQuestionViewModel = viewModel.surveyQuestionViewModel
                        navigator.show(
                            screen: .surveyQuestion(viewModel: surveyQuestionViewModel),
                            by: .presentCover
                        )
```

Comment (markgravity) [reply ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/SurveyDetailScreen.swift]: @suho updated at [279d78c](https://github.com/markgravity/kmm-ic/pull/88/commits/279d78ce7c4f986afed3b97d8f4e79fab139fd9c)

File: ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/Models/SurveyDetailUIModel.swift (hunk 1 above)
Comment (markgravity) [reply ios/KMMIC/Sources/Presentation/Modules/SurveyDetail/Models/SurveyDetailUIModel.swift]: @suho updated at [279d78c](https://github.com/markgravity/kmm-ic/pull/88/commits/279d78ce7c4f986afed3b97d8f4e79fab139fd9c)

**Review by suho — APPROVED** [review]

**Review by phongvhd93 — APPROVED** [review]

**Pull request merged**